keyboards/gmmk/pro/rev2/ansi/ansi.c

```diff
@@ -0,0 +1,265 @@
+    {0, CS7_SW12, CS8_SW12, CS9_SW12},     //  63, [, k81
+    {0, CS10_SW12, CS11_SW12, CS12_SW12},  //  64, ", k83
+    {0, CS16_SW12, CS17_SW12, CS18_SW12},  //  65, Ct_R, k04
+
+    {1, CS1_SW1, CS2_SW1, CS3_SW1},     //  66, F12, ka5
+    {1, CS13_SW1, CS14_SW1, CS15_SW1},  //  67, LED, l01
+    {1, CS16_SW1, CS17_SW1, CS18_SW1},  //  68, LED, l11
+    {1, CS4_SW2, CS5_SW2, CS6_SW2},     //  69, Prt, k97
+    {1, CS13_SW2, CS14_SW2, CS15_SW2},  //  70, LED, l02
+    {1, CS16_SW2, CS17_SW2, CS18_SW2},  //  71, LED, l12
+    {1, CS4_SW3, CS5_SW3, CS6_SW3},     //  72, Del, k65
+    {1, CS13_SW3, CS14_SW3, CS15_SW3},  //  73, LED, l03
+    {1, CS16_SW3, CS17_SW3, CS18_SW3},  //  74, LED, l13
+    {1, CS4_SW4, CS5_SW4, CS6_SW4},     //  75, PgUp, k15
+    {1, CS13_SW4, CS14_SW4, CS15_SW4},  //  76, LED, l04
+    {1, CS16_SW4, CS17_SW4, CS18_SW4},  //  77, LED, l14
+    {1, CS1_SW5, CS2_SW5, CS3_SW5},     //  78, =, k66
+    {1, CS10_SW5, CS11_SW5, CS12_SW5},  //  79, Right, k05
+    {1, CS13_SW5, CS14_SW5, CS15_SW5},  //  80, LED, l05
+    {1, CS16_SW5, CS17_SW5, CS18_SW5},  //  81, LED, l15
+    {1, CS4_SW6, CS5_SW6, CS6_SW6},     //  82, End, k75
+    {1, CS13_SW6, CS14_SW6, CS15_SW6},  //  83, LED, l06
+    {1, CS16_SW6, CS17_SW6, CS18_SW6},  //  84, LED, l16
+    {1, CS1_SW7, CS2_SW7, CS3_SW7},     //  85, BSpc, ka1
+    {1, CS4_SW7, CS5_SW7, CS6_SW7},     //  86, PgDn, k25
+    {1, CS13_SW7, CS14_SW7, CS15_SW7},  //  87, LED, l07
+    {1, CS16_SW7, CS17_SW7, CS18_SW7},  //  88, LED, l17
+    {1, CS1_SW8, CS2_SW8, CS3_SW8},     //  89, ], k61
+    {1, CS4_SW8, CS5_SW8, CS6_SW8},     //  90, Sh_R, k91
+    {1, CS13_SW8, CS14_SW8, CS15_SW8},  //  91, LED, l08
+    {1, CS16_SW8, CS17_SW8, CS18_SW8},  //  92, LED, l18
+    {1, CS1_SW9, CS2_SW9, CS3_SW9},     //  93, \, ka2
+    {1, CS4_SW9, CS5_SW9, CS6_SW9},     //  94, Up, k35
+    {1, CS4_SW10, CS5_SW10, CS6_SW10},  //  95, Left, k03
+    {1, CS1_SW11, CS2_SW11, CS3_SW11},  //  96, Enter, ka4
+    {1, CS4_SW11, CS5_SW11, CS6_SW11},  //  97, Down, k73
+};
+// clang-format on
+#endif
+
+#ifdef EEPROM_ENABLE
+
+#include "spi_master.h"
+
+void spi_init(void) {
+    static bool is_initialised = false;
+    if (!is_initialised) {
+        is_initialised = true;
+
+        // Try releasing special pins for a short time
+        setPinInput(SPI_SCK_PIN);
+        setPinInput(SPI_MOSI_PIN);
+        setPinInput(SPI_MISO_PIN);
+
+        chThdSleepMilliseconds(10);
+
+        palSetPadMode(PAL_PORT(SPI_SCK_PIN), PAL_PAD(SPI_SCK_PIN), PAL_MODE_ALTERNATE(SPI_SCK_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST | PAL_WB32_CURRENT_LEVEL3);
+        palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), PAL_MODE_ALTERNATE(SPI_MOSI_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+        palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), PAL_MODE_ALTERNATE(SPI_MISO_PAL_MODE) | PAL_OUTPUT_TYPE_PUSHPULL | PAL_OUTPUT_SPEED_HIGHEST);
+    }
+}
```

Comment:
Is this still needed?

Reply:
Please check out this PR. #17534

Reply:
I mean, with the changes made in #17650 is this override necessary?

Reply:
@tzarc  Not necessary.

Reply:
Let me remove the override first, and you can agree to merge after #17534 is resolved.